modules/tensor_mechanics/include/materials/CappedMohrCoulombCosseratStressUpdate.h

```diff
@@ -0,0 +1,71 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#ifndef CAPPEDMOHRCOULOMBCOSSERATSTRESSUPDATE_H
+#define CAPPEDMOHRCOULOMBCOSSERATSTRESSUPDATE_H
+
+#include "CappedMohrCoulombStressUpdate.h"
+
+class CappedMohrCoulombCosseratStressUpdate;
+
+template <>
+InputParameters validParams<CappedMohrCoulombCosseratStressUpdate>();
+
+/**
+ * CappedMohrCoulombCosseratStressUpdate implements rate-independent nonassociative
+ * Mohr-CoulombC plus tensile plus compressive plasticity with hardening/softening
+ * in the Cosserat setting.  The Mohr-Coulomb plasticity is based on the symmetric
+ * part of the stress tensor only, and uses an isotropic elasticity tensor.
+ */
+class CappedMohrCoulombCosseratStressUpdate : public CappedMohrCoulombStressUpdate
+{
+public:
+  CappedMohrCoulombCosseratStressUpdate(const InputParameters & parameters);
+
+  bool requiresIsotropicTensor() override { return false; }
+
+protected:
+  /// Young's modulus of the host material
+  const Real _host_young;
+
+  /// Poisson's of the host material
+  const Real _host_poisson;
+
+  /// E0011 = Lame lambda modulus of the host material
+  const Real _host_E0011;
+
+  /// E0000 = Lame lambda + 2 * shear modulus of the host material
+  const Real _host_E0000;
+
+  virtual void setStressAfterReturnV(const RankTwoTensor & stress_trial,
+                                     const std::vector<Real> & stress_params,
+                                     Real gaE,
+                                     const std::vector<Real> & intnl,
+                                     const yieldAndFlow & smoothed_q,
+                                     const RankFourTensor & Eijkl,
+                                     RankTwoTensor & stress) const override;
+
+  virtual void preReturnMapV(const std::vector<Real> & trial_stress_params,
```

Comment:
In my mind it stands for "Vector", since all these "V" classes work on std::vectors of things.  I can change if you want, but it needs me to change a whole bunch of classes (which isn't hard of course) but because of that i'll wait for your direction until i make the change.